src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] list;
+    private int size;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (checkCapacity()) {
+            resize();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
let's check if index is valid
the logic is similar to that in your `checkIndexBounds()` method, but not the same - `index` may be equal to `size`